### ISSUE TITLE
Discussion of DSL for Generamba 2.0

### ISSUE THREAD

**Issue by etolstoy** [rambler-digital-solutions/Generamba#164]
We are happy to announce that we've started working on the next big release of Generamba together with @rambler-digital-solutions/ruby-team 🎉🎉🎉. We'll publish a full roadmap soon - but for now, the next step is switching from `yml` syntax to a custom DSL for `Rambafile`. It'll allow us to simplify the generation process, support multiple `gen` configurations and allow to use complex chains of custom actions. The possibilities are literally endless :)

[Here](https://github.com/rambler-digital-solutions/Generamba/wiki/Generamba-2.0-DSL-Syntax) you can have a look at the example syntax of new `Rambafile`. If you have any thoughts and ideas, feel free to post your comments in this issue.

**Comment by MaximBazarov** [comment]
What about option to chaining  templates, for instance:
generateTemplateA(someparameters: "...")
than 
generateTemplateB(someparameters: "...")

so first generates template files and in this falser also put files from second template?

**Comment by etolstoy** [comment]
@MaksimBazarov It's already prototyped:

```
  gen :module1 do |options|
    options[:template_name] = 'Module1_template'
    options[:custom_parameter1] = true
  end

  gen :module2 do |options|
    options[:template_name] = 'Module2_template'
    options[:custom_parameter1] = false
  end
```

**Comment by etolstoy** [comment]
We've published a full roadmap for 2.0-alpha: https://github.com/rambler-digital-solutions/Generamba/blob/develop/docs/2.x/roadmap.md.
